pymc3/step_methods/slicer.py

```diff
@@ -61,36 +62,40 @@ def __init__(self, vars=None, w=1.0, tune=True, model=None, iter_limit=np.inf, *
         super().__init__(vars, [self.model.fastlogp], **kwargs)
 
     def astep(self, q0, logp):
-        self.w = np.resize(self.w, len(q0))  # this is a repmat
-        q = np.copy(q0)  # TODO: find out if we need this
-        ql = np.copy(q0)  # l for left boundary
-        qr = np.copy(q0)  # r for right boudary
-        for i in range(len(q0)):
+        q0_val = q0.data
+        self.w = np.resize(self.w, len(q0_val))  # this is a repmat
+        q = np.copy(q0_val)  # TODO: find out if we need this
+        ql = np.copy(q0_val)  # l for left boundary
+        qr = np.copy(q0_val)  # r for right boudary
+        for i in range(len(q0_val)):
             # uniformly sample from 0 to p(q), but in log space
-            y = logp(q) - nr.standard_exponential()
+            q_ra = RaveledVars(q, q0.point_map_info)
+            y = logp(q_ra) - nr.standard_exponential()
             ql[i] = q[i] - nr.uniform(0, self.w[i])
             qr[i] = q[i] + self.w[i]
             # Stepping out procedure
             cnt = 0
-            while y <= logp(ql):  # changed lt to leq  for locally uniform posteriors
+            while y <= logp(
+                RaveledVars(ql, q0.point_map_info)
+            ):  # changed lt to leq  for locally uniform posteriors
                 ql[i] -= self.w[i]
                 cnt += 1
                 if cnt > self.iter_limit:
                     raise RuntimeError(LOOP_ERR_MSG % self.iter_limit)
             cnt = 0
-            while y <= logp(qr):
+            while y <= logp(RaveledVars(qr, q0.point_map_info)):
                 qr[i] += self.w[i]
                 cnt += 1
                 if cnt > self.iter_limit:
                     raise RuntimeError(LOOP_ERR_MSG % self.iter_limit)
 
             cnt = 0
             q[i] = nr.uniform(ql[i], qr[i])
-            while logp(q) < y:  # Changed leq to lt, to accomodate for locally flat posteriors
+            while logp(q_ra) < y:  # Changed leq to lt, to accomodate for locally flat posteriors
```

Comment:
⚠ before this line used `q`, which was inplace-edited in the line above.
If and only if `RaveledVars` maintains the pointer to the underlying `ndarray`, the `RaveledVars` objects should be created all right after the `copy()` lines (L69).

Reply:
A `RaveledVars` object is just a named tuple, so, yes, it holds a reference to `q` and reflects any in-place changes made to `q`.